dev/scripts/src/alluxio.org/build/cmd/build.go

```diff
@@ -138,7 +138,11 @@ func buildTarball(opts *buildOpts) error {
 
 	} else {
 		log.Printf("Running maven command from %v:\n%v", repoBuildDir, baseMvnCmd)
-		cmd := command.New(baseMvnCmd).WithDir(repoBuildDir)
+		commandToRun := baseMvnCmd
+		if opts.skipWebUi {
+			commandToRun = baseMvnCmd + " " + "-pl '!webui'"
```

Comment:
i don't like how this adds in the `-pl` flag forcibly. the tarball profile can optionally provide maven args via the `mvnArgs` field. if this field includes a `-pl` option, then the maven command will fail. ex. `mvn install -am -pl '!webui' -pl 'common/transport'` doesn't work. but i also don't have a good suggestion to handle the string because the `-pl` could be provided in various formats.

a more formal way to introduce better control over `-pl` is to add a new string parameter for `mavenProjectList` such that there is only one allowed place to provide values for this argument. we would explicitly disallow the substring `-pl` in `mvnArgs`. then this section would look something like
```
if opts.mavenProjectList != "" {
	commandToRun = baseMvnCmd + fmt.Sprintf(" -pl '%s'", opts.mavenProjectList)
}
```

Reply:
Yeah, i try "mvnArgs" and it fails because exactly the same reason